scipy/interpolate/_cubic.py

```diff
@@ -330,3 +337,204 @@ def from_spline(cls, tck, extrapolate=None):
+            raise ValueError("`y` must contain only finite values.")
+
+        dx = np.diff(x)
+        if np.any(dx <= 0):
+            raise ValueError("`x` must be strictly increasing sequence.")
+
+        # Find derivative values at each x[i] by solving a tridiagonal system.
+        # If n=2 or n=3, "not-a-knot" condition is not enough to determine
+        # derivatives. In this case the solution is chosen to be a
+        # linear/quadratic function passing through the given points.
+        n = x.shape[0]
+        y = np.rollaxis(y, axis)
+        dxr = dx.reshape((dx.shape[0],) + (1,)*(y.ndim-1))
+        slope = np.diff(y, axis=0) / dxr
+        if n == 2:
+            s = np.empty((2,) + y.shape[1:], dtype=y.dtype)
+            s[0] = slope
+            s[1] = slope
+        elif n == 3:
+            A = np.zeros((3, 3))  # This is a standard matrix.
+            b = np.empty((3,) + y.shape[1:], dtype=y.dtype)
+
+            A[0, 0] = 1
+            A[0, 1] = 1
+            A[1, 0] = dx[1]
+            A[1, 1] = 2 * (dx[0] + dx[1])
+            A[1, 2] = dx[0]
+            A[2, 1] = 1
+            A[2, 2] = 1
+
+            b[0] = 2 * slope[0]
+            b[1] = 3 * (dxr[0] * slope[1] + dxr[1] * slope[0])
+            b[2] = 2 * slope[1]
+
+            s = solve(A, b, overwrite_a=True, overwrite_b=True,
+                      check_finite=False)
+        else:
+            A = np.zeros((3, n))  # This is a banded matrix representation.
+            b = np.empty((n,) + y.shape[1:], dtype=y.dtype)
+
+            A[1, 0] = dx[1]
+            A[1, -1] = dx[-2]
+            A[1, 1:-1] = 2 * (dx[:-1] + dx[1:])
+            A[0, 1] = x[2] - x[0]
+            A[0, 2:] = dx[:-1]
+            A[-1, -2] = x[-1] - x[-3]
+            A[-1:, :-2] = dx[1:]
+
+            d = x[2] - x[0]
+            b[0] = ((dxr[0] + 2*d)*dxr[1]*slope[0] + dxr[0]**2*slope[1]) / d
+            b[1:-1] = 3 * (dxr[1:] * slope[:-1] + dxr[:-1] * slope[1:])
+            d = x[-1] - x[-3]
+            b[-1] = ((dxr[-1]**2*slope[-2] +
+                      (2*d + dxr[-1])*dxr[-2]*slope[-1]) / d)
+
+            s = solve_banded((1, 1), A, b, overwrite_ab=True, overwrite_b=True,
+                             check_finite=False)
+
+        # Compute coefficients in PPoly form.
+        c0 = (s[:-1] + s[1:] - 2 * slope) / dxr**2
+        c = np.empty((4, n - 1) + y.shape[1:], dtype=c0.dtype)
+        c[0] = c0
+        c[1] = (slope - s[:-1]) / dxr - c0 * dxr
+        c[2] = s[:-1]
+        c[3] = y[:-1]
+
+        super(CubicSpline, self).__init__(c, x, extrapolate=extrapolate)
+        self.axis = axis
+
+    def extend(self, c, x, right=True):
+        raise NotImplementedError("Extending a CubicSpline is not yet "
+                                  "implemented.")
+
+    @classmethod
+    def from_spline(cls, tck, extrapolate=None):
+        raise NotImplementedError("Conversion from B-spline representation is "
+                                  "not yet implemented.")
```

Comment:
I thought that somehow this method should restore the original `x` as breakpoints, now I see that it's impossible. So I'll remove this.